src/plugins/unified_field_list/public/hooks/use_memorized_date_range.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import dateMath from '@kbn/datemath';
+import { useMemo } from 'react';
+
+export const useMemorizedDateRange = (
+  fromDate: string,
+  toDate: string
+): { from: string; to: string } => {
+  return useMemo(() => {
+    const roundedDateRange = {
+      from: dateMath.parse(fromDate)!.startOf('minute').toISOString(),
+      to: dateMath.parse(toDate, { roundUp: true })!.endOf('minute').toISOString(),
+    };
+
+    return roundedDateRange;
+  }, [fromDate, toDate]);
+};
```

Comment:
Here's something we should discuss, because it also changes Lens behavior dear @stratoula & @flash1293(⚠️)  

In Discover, before this PR, after fetching the field list, there were no follow up requests, since the field existence was calculated based on the returned.

In Lens, this was different, every refetch triggers a request to get those fields.

So now with aligning both apps, the same behavior when there's a constantly changing absolute time range leads to the loop of render side bar, see new absolute range, request data, rerender side bar, see new absolute range and so on. This can lead to a lot of redundant requests, however, there are also cases where new fields are of interest and are added to the field list quickly. 

The current change aligns to the behavior of our autocomplete functionality, which caches the requested time range requests by minute.

This would mean that without any changes to the query/filter/dataview, there would be only a max request to ES once a minute.

So the question is, are we fine with a new behavior like this? And would 1 min be the value we should choose for that?

Reply:
Very helpful info! Thanks, @flash1293 !